### PR TITLE
CATTY-294 Split methods in FormulaEditorViewController

### PR DESCRIPTION
**Pull request by DhrubojyotiBis1**
The following functions are modified:

- addNewVarOrList -> addNewVariable and addNewList
- askForVariableName -> askForVariableName and askForListName
- saveVariable -> saveVariable and saveList
- isVarOrListBeingUsed -> isVariableUsed and isListUsed

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [ ] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: src/Catty/FormulaEditor/ViewController/FormulaEditorViewController.m [hunk 1]
```diff
@@ -1085,31 +1135,13 @@ - (IBAction)deleteVariable:(UIButton *)sender {
             }
```
Comment (m-herold) [inline src/Catty/FormulaEditor/ViewController/FormulaEditorViewController.m:1135]: Can you please try to also rename this method, for example to `deleteUserData`?

File: src/Catty/FormulaEditor/ViewController/FormulaEditorViewController.m [hunk 2]
```diff
@@ -1085,31 +1135,13 @@ - (IBAction)deleteVariable:(UIButton *)sender {
...
         }
         if (userVariable) {
-            if(![self isVarOrListBeingUsed:userVariable]) {
+            if(![self isVariableUsed:userVariable] || ![self isListUsed:userVariable]) {
```
Comment (m-herold) [inline src/Catty/FormulaEditor/ViewController/FormulaEditorViewController.m:1138]: Please have a closer look at this statement. `isVariableUsed` should only be called if it is a variable, and `isListUsed` only if it is a list.

File: src/Catty/FormulaEditor/ViewController/FormulaEditorViewController.m [hunk 3]
```diff
@@ -1118,9 +1150,39 @@ - (IBAction)deleteVariable:(UIButton *)sender {
...
+        }
+        [self.object.project saveToDiskWithNotification:YES];
+        [self updateVariablePickerData];
+    }
+}
+
+- (void)deleteList: (id<UserDataProtocol>)userList atrow:(NSInteger)row isProjectData:(BOOL)isProjectData
 {
-    if([self.object.project.variables isProjectVariable:variable] || [self.object.project.variables isProjectList:variable]) {
+    BOOL removed = NO;
```
Comment (m-herold) [inline src/Catty/FormulaEditor/ViewController/FormulaEditorViewController.m:1170]: Please assign directly to variable (merge those two lines).

File: src/Catty/FormulaEditor/ViewController/FormulaEditorViewController.m [hunk 4]
```diff
@@ -1118,9 +1150,39 @@ - (IBAction)deleteVariable:(UIButton *)sender {
     }
 }
 
-- (BOOL)isVarOrListBeingUsed:(UserVariable*)variable
+- (void)deleteVariable: (UserVariable*)userVariable atrow:(NSInteger)row isProjectData:(BOOL)isProjectData
+{
+    BOOL removed = NO;
```
Comment (m-herold) [inline src/Catty/FormulaEditor/ViewController/FormulaEditorViewController.m:1155]: Please assign directly to variable (merge those two lines).

**Comment by DhrubojyotiBis1** [comment]
Hi, @m-herold I have made the changes. Please review

**Comment by DhrubojyotiBis1** [comment]
Hi, @m-herold I have made the changes. Please review

File: src/Catty/Defines/LanguageTranslationDefines.h [hunk 5]
```diff
@@ -579,6 +579,7 @@
 #define kUIFEProjectLists NSLocalizedString(@"Project lists:", nil)
 #define kUIFEObjectLists NSLocalizedString(@"Object lists:", nil)
 #define kUIFEDeleteVarBeingUsed NSLocalizedString(@"This variable can not be deleted because it is still in use.", nil)
+#define kUIFEDeleteListBeingUsed NSLocalizedString(@"This List can not be deleted because it is still in use.", nil)
```
Comment (m-herold) [inline src/Catty/Defines/LanguageTranslationDefines.h:582]: Please change to "This list" (lowercase).

**Pull request merged**